### PR TITLE
[CRM457-2282] additional main offence type question and feature flag

### PR DESCRIPTION
**Pull request by robinsk1**
## Description of change

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-2282)

- [x] [requires change to schema - merge common forms gem PR and update gemfile before merging this one](https://github.com/ministryofjustice/laa-crime-forms-common/pull/19)
- [x] bump app-store, CW app to use new gem version
- [ ] [fix e2e tests (offence type required)](https://github.com/ministryofjustice/nsm-e2e-test/pull/172)

## Notes for reviewer

Do we have to backfill this? 

## Screenshots of changes (if applicable)

Screenshot 2024-11-21 at 14 33 04

### CODE REVIEW

File: app/forms/nsm/steps/case_details_form.rb [hunk 1]
```diff
@@ -17,6 +18,7 @@ class CaseDetailsForm < ::Steps::BaseFormObject
                 presence: true,
                 multiparam_date: { allow_past: true, allow_future: false },
                 if: :remitted_to_magistrate?
+      validates :main_offence_type, presence: true
```
Comment (elken) [inline app/forms/nsm/steps/case_details_form.rb:21]: Should also have a validation for inclusion

```ruby
validates_inclusion_of :main_offence_type, in: MainOffenceType.values
```

**Review by elken — APPROVED** [review]

File: app/views/nsm/steps/case_details/edit.html.erb [hunk 2]
```diff
@@ -8,6 +8,14 @@
     <%= step_form @form_object do |f| %>
       <%= suggestion_select f, :main_offence, MainOffence.all, :name, :name, 'offence-autocomplete', width: 20,
                               options: { include_blank: true }, data: { autoselect: false }, label: { size: 's' } %>
+
+      <%= f.govuk_radio_buttons_fieldset :main_offence_type, legend: { size: 's', class: 'govuk-!-margin-bottom-6' } do %>
+        <% MainOffenceType.values.each_with_index do |main_offence_type, index| %>
```
Comment (ivanELEC) [inline app/views/nsm/steps/case_details/edit.html.erb:13]: This isn't to spec (re: what html components it adds, see this [comment](https://github.com/ministryofjustice/laa-submit-crime-forms/pull/1310#discussion_r1852241256))

File: app/presenters/nsm/check_answers/case_details_card.rb [hunk 3]
```diff
@@ -19,6 +19,10 @@ def row_data
             head_key: 'main_offence',
             text: check_missing(claim.main_offence)
           },
+          {
+            head_key: 'main_offence_type',
+            text: check_missing(claim.main_offence_type)
+          },
           {
```
[the remark this answers is not in the document]
Comment (ivanELEC) [reply app/presenters/nsm/check_answers/case_details_card.rb:26]: actually never mind this comment at all, makes more sense for this to be here as this isn't feature flagged

File: app/views/nsm/steps/case_details/edit.html.erb (hunk 2 above)
Comment (elken) [reply app/views/nsm/steps/case_details/edit.html.erb:13]: That's only for when it's the only question on the page AFAICT

Comment (ivanELEC) [reply app/views/nsm/steps/case_details/edit.html.erb:13]: true sorry

**Review by ivanELEC — APPROVED** [review]

**Pull request merged**